diff --git a/roller/05_server-cas.py b/roller/05_server-cas.py
index 46a46ce..2948d37 100644
--- a/roller/05_server-cas.py
+++ b/roller/05_server-cas.py
@@ -24,7 +24,7 @@ write_api = client.write_api(write_options=SYNCHRONOUS)
 # Cassandra settings
 cassandra_hosts = ["127.0.0.1"]
 cassandra_keyspace = "bfh"
-cassandra_table = "light_readings"
+cassandra_table = "sensor"
 
 cassandra_cluster = Cluster(cassandra_hosts)
 cassandra_session = cassandra_cluster.connect()
@@ -42,18 +42,17 @@ cassandra_session.execute(
     CREATE TABLE IF NOT EXISTS {cassandra_table} (
         sensor_id text,
         day_bucket date,
-        ts_ns bigint,
         ts timestamp,
         lx double,
-        PRIMARY KEY ((sensor_id, day_bucket), ts_ns)
-    ) WITH CLUSTERING ORDER BY (ts_ns ASC)
+        PRIMARY KEY ((sensor_id, day_bucket), ts)
+    ) WITH CLUSTERING ORDER BY (ts ASC)
     """
 )
 
 insert_cassandra_stmt = cassandra_session.prepare(
     f"""
-    INSERT INTO {cassandra_table} (sensor_id, day_bucket, ts_ns, ts, lx)
-    VALUES (?, ?, ?, ?, ?)
+    INSERT INTO {cassandra_table} (sensor_id, day_bucket, ts, lx)
+    VALUES (?, ?, ?, ?)
     """
 )
 
@@ -62,18 +61,17 @@ def generator():
     data = json.loads(request.data)
     print(data)
 
-    id = "light"
+    id = data.get("id", "light")
     time_sec = float(data["time_sec"])
     lx = float(data["lx"])
 
-    ts_ns = int(time_sec * 1_000_000_000)
     ts_dt = datetime.fromtimestamp(time_sec, tz=timezone.utc)
     day_bucket = ts_dt.date()
 
     try:
         point = influxdb_client.Point(id) \
             .field("lx", lx) \
-            .time(ts_ns, write_precision='ns')
+            .time(int(time_sec * 1_000_000_000), write_precision='ns')
         write_api.write(bucket=bucket, org=org, record=point)
     except Exception as e:
         print(f"Error writing to InfluxDB: {e}")
@@ -81,9 +79,7 @@ def generator():
     try:
         cassandra_session.execute(
             insert_cassandra_stmt,
-            # id und day_bukcket bilden den Partition Key
-            # querys wie alle daten für ein tag werden schnell
-            (id, day_bucket, ts_ns, ts_dt, lx),
+            (id, day_bucket, ts_dt, lx),
         )
     except Exception as e:
         print(f"Error writing to Cassandra: {e}")
